File: server/models/elastic/listing.py
from elasticsearch import Elasticsearch
from elasticsearch_dsl import Document, Text, Integer, Keyword, Float, Object, analyzer, connections


class Listing(Document):
    identifier = Integer()
    name = Text()
    url = Text()
    price = Float()
    category = Text(fielddata=True)
    image = Text()
    properties = Object()
    source_id = Integer()
    source_name = Keyword()
    source_url = Keyword()

    class Index:
        name = "listings"
        settings = {
            "number_of_shards": 2
        }


Listing.init()

File: client/src/types/Listing.ts

export interface IListing {
	identifier: number;
	name: string;
	url: string;
	price?: number;
	category?: string;
	image: string;
	properties: {
		description: string;
	};
	sourceName: string;
}

export class Listing implements IListing {
	identifier: number;
	name: string;
	url: string;
	price?: number;
	category?: string;
	image: string;
	sourceName: string;
	properties: {
		description: string;
	};

	constructor(listing: IListing) {
		this.identifier = listing.identifier;
		this.name = listing.name;
		this.url = listing.url;
		this.price = listing.price;
		this.category = listing.category;
		this.image = listing.image;
		this.properties = listing.properties;
		this.sourceName = listing.sourceName;
	}

	getDescription(): string {
		return this.properties.description.split(". ").join("\n");
	}
}
